primitives/src/validator.rs

```diff
@@ -48,6 +48,14 @@ impl TryFrom<&str> for ValidatorId {
         let result = hex::decode(hex_value).map_err(|_| {
             DomainError::InvalidArgument("Failed to deserialize validator id".to_string())
         })?;
+
+        if result.len() != 20 {
```

Comment:
please, don't do this multiple times. We already have a check above.
`hex_value` can be Result, we can check first for `42` then `40` else do `Err` and at the end just `?`.
This will remove the split that is currently here and make it consistent.

Reply:
Ok, lets leave it like that.